src/webapi/node.rs

```diff
@@ -362,6 +362,58 @@ pub struct Node( Reference );
 impl IEventTarget for Node {}
 impl INode for Node {}
 
+impl Node {
+    /// Attempt to create the `Node` from raw html.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if the resulting number of nodes != 1. To ensure this
+    /// it is recommended to at _least_ have programatic control of your root
+    /// HTML node.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let node = Node::from_html("<div>Some text, horray!</div>").unwrap();
+    /// ```
+    pub fn from_html(html: &str) -> Result<Node, SyntaxError> {
+        let result = js! {
+            var span = document.createElement("span");
+            try {
+                span.innerHTML = @{html};
+            } catch( error ) {
+                return {
+                    error: error,
+                    success: false,
+                };
+            }
+
+            if( span.childNodes.length != 1 ) {
+                return {
+                    panic: "HTML had ${span.childNodes.length} nodes but must have 1",
+                    success: false,
+                };
+            }
+            return span.childNodes[0];
+        };
+
+
+        if let Value::Bool(false) = js!{ return @{&result}.success; } {
+            if let Value::String(e) = js! { return @{&result}.panic; } {
+                panic!(e);
```

Comment:
Hmmm... do we actually want to panic here?

I mean - putting a snippet of HTML here with multiple roots is *kind-of* a syntax error, so perhaps we should just return `SyntaxError` in such a case with our own error message?

It might not make much sense to treat the case where the HTML has an actual syntax error and the case where the HTML has multiple roots separately. *Both* cases should (in theory) never happen if we control the HTML (assuming we didn't make a mistake), and both *can* happen if we just paste the HTML from an external source.

And in general I don't think we currently don't have any other panics like this one anywhere, so it'd would be a first.

Reply:
I think that reasoning makes sense. I guess I just create it [using the js! macro with SyntaxError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SyntaxError).

I'm kind of concerned that such an error gets created for number of nodes but is NOT created if the html is invalid (see above examples) -- but I guess that is just the harm of working with javascript!

Reply:
Well, yeah, if your document is HTML then the snippet you pass to `innerHTML` is treated very loosely. That's an unfortunate historic artifact.

Reply:
I mean, we could always validate it as valid html _in the future_ if such a thing were useful (I think it would be).

Reply:
I thought about it too, but I'm not sure whenever something like that would pull its own weight. Maybe in debug mode? A full HTML parser for this little snippet seems like an overkill. (Alternatively we could use the `document.implementation.createDocument(null, null);` trick to create a dummy XHTML document and abuse the browser to do the validation for us, even where our document is HTML.)

Reply:
Just to have a reference, that would be:

```
var xmldoc = document.implementation.createDocument(null, null);
var span = xmldoc.createElement("span"); 
span.innerHTML = "<div"; // Will throw SyntaxError
```
Seems hacky, but I much prefer requiring valid syntax. So sounds good.

Reply:
Yeah, something like that; I think that would be fine to have in debug mode.